### PR TITLE
Related to #144

### PR DESCRIPTION
**Pull request by siad007**
Would fix various uri behavior according to rfc 3986 instead of only support rfc 2396.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,5 +5,6 @@ coverage
 .buildpath
 .project
 .settings
+.idea
```
Comment (bighappyface) [inline .gitignore]: This should be added to a [global ignore](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore)

**Comment by bighappyface** [comment]
I prefer a more descriptive commit message

File: .gitignore (hunk 1 above)
Comment (siad007) [inline .gitignore]: @bighappyface I would suggest to remove
```
.buildpath
.project
.settings
```
as well.

**Comment by bighappyface** [comment]
@siad007 thanks for updates. Next steps are to rebase and squash this down to a single commit.

**Comment by bighappyface** [comment]
Closed due to age

**Pull request closed without merging**